src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -231,10 +221,19 @@ export class Kernel implements IKernel {
                     }
                     resolve(this.notebook);
                 } catch (ex) {
+                    sendKernelTelemetryEvent(
+                        options.document.uri,
+                        Telemetry.NotebookStart,
+                        stopWatch.elapsedTime,
+                        undefined,
+                        ex
+                    );
                     if (options.disableUI) {
                         sendTelemetryEvent(Telemetry.KernelStartFailedAndUIDisabled);
+                    } else {
+                        this.errorHandler.handleError(ex).ignoreErrors(); // Just a notification, so don't await this
                     }
-                    traceError('failed to start INotebook in kernel', ex);
+                    traceError(`failed to start INotebook in kernel, UI Disabled = ${options.disableUI}`, ex);
```

Comment:
Included more information to determine when this failed (if UI interaction was allowed or not)